### PR TITLE
chore: migrate to setup.cfg

### LINKED ISSUE #644: Need bump pure-sasl version to support winkerberos
Now this project's setup.py hardcodes pure-sasl to 0.5.1. https://github.com/python-zk/kazoo/blob/6337fd6f72b59fb20886f980f2e0d6d41525dc35/setup.py#L80

But pure-sasl supports winkerberos on Windows since its version 0.6.0. https://github.com/thobbs/pure-sasl/blob/master/CHANGES.rst#060

So it's good to bump pure-sasl version requirement to 0.6. However there is still an issue in pure-sasl that it does not support winkerberos 0.8, I created a ticket there https://github.com/thobbs/pure-sasl/pull/34 And hereby I create this ticket as a memo: so once pure-sasl has released that change, we'd better bump requirement in this kazoo project.

### PR DESCRIPTION
**Pull request by ceache**
closes #644 #662 #663

## Why is this needed?

Migrate to python packaging best practices.

## Proposed Changes

  - use setup.cfg & pyproject.toml to build kazoo

## Does this PR introduce any breaking change?

No

### CODE REVIEW

**Comment by StephenSorriaux** [comment]
Hey @ceache 

Really sorry for not having check this before :(

File: setup.cfg [hunk 1]
```diff
@@ -1,15 +1,95 @@
...
+keywords = zookeeper, lock, leader, configuration
+classifiers =
+    Development Status :: 5 - Production/Stable
+    License :: OSI Approved :: Apache Software License
+    Intended Audience :: Developers
+    Operating System :: OS Independent
+    Programming Language :: Python
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
```
Comment (StephenSorriaux) [inline setup.cfg:22]: is Python 3.9 missing?

**Review by StephenSorriaux — APPROVED** [review]
Thank you!

**Comment by ceache** [comment]
Wasn't in scope originally but I'll add it in a next PR

**Pull request merged**